corehq/util/decorators.py

```diff
@@ -62,3 +64,69 @@ def __enter__(self):
 
     def __exit__(self, exc_type, exc_val, exc_tb):
         pass
+
+
+class CouldNotAqcuireLock(Exception):
+    pass
+
+
+# Sorry this is so magic
+def _get_unique_key(format_str, fn, *args, **kwargs):
+    """
+    Lines args and kwargs up with those specified in the definition of fn and
+    passes the result to `format_str.format()`.
+    """
+    varnames = fn.func_code.co_varnames
+    kwargs.update(dict(zip(varnames, args)))
+    return ("{}-" + format_str).format(fn.__name__, **kwargs)
+
+
+def locking_task(unique_key, default_retry_delay=30, timeout=5*60, max_retries=3):
+    """
+    Define a locking celery task to prevent race conditions.
```

Comment:
i think we want to clarify this - if we just wanted to prevent race conditions we'd use the `CriticalSection`. the use case here was to create serially-executed tasks